frontend/components/AddFundsThroughBridge/useGenerateAddFunds.tsx

```diff
@@ -0,0 +1,126 @@
+import { uniq } from 'lodash';
+import { useCallback, useMemo, useState } from 'react';
+
+import { AddressBalanceRecord } from '@/client';
+import { AddressZero } from '@/constants/address';
+import { TokenSymbol } from '@/enums/Token';
+import { useServices } from '@/hooks/useServices';
+import { Address } from '@/types/Address';
+import { BridgeRequest } from '@/types/Bridge';
+import { typedKeys } from '@/types/Util';
+import { getTokenDetailsFromAddress } from '@/utils/middlewareHelpers';
+
+import { DefaultTokenAmount } from './types';
+import { useAddFundsGetBridgeRequirementsParams } from './useAddFundsGetBridgeRequirementsParams';
+
+export type GeneratedInput = {
+  tokenAddress: Address;
+  symbol: TokenSymbol;
+  amount: number;
+};
+
+/**
+ * Generate inputs to add funds for bridging.
+ *
+ * Example: If the chain is Gnosis, user can add OLAS and XDAI.
+ * This function will return an array of objects with tokenAddress, symbol, and amount set to 0.
+ */
+export const useGenerateAddFunds = ({
+  requirements,
+  defaultTokenAmounts,
+  destinationAddress,
+  onlyNativeToken = false,
+}: {
+  requirements: AddressBalanceRecord;
+  defaultTokenAmounts?: DefaultTokenAmount[];
+  destinationAddress?: Address;
+  onlyNativeToken?: boolean;
```

Comment:
maybe we could do `displayedTokens: symbols[]` instead, and use it as some sort of a config? in case in future we'd need to bridge only let's say USDC, would be very easy to reuse